### PR TITLE
Column Transformation changes

### PR DESCRIPTION
**Pull request by davidboren**
Here's the column transformation changes we talked about.  The version incrementation was just for the benefit of my lockfile management.  I made up a new graph using median time instead of mean.  Took longer but is a bit cleaner.  Everything looks as we'd expect.  Slight overhead from the code additions, but much much faster when we assign to column level inputs.  Ooooo-rah
 
alt text

Used the identity function for the column transformation when not assigning to inputs.  For assigning to inputs I used a pretty normal-looking base-case:

```R
ct <- column_transformation(function(x) {
  if ('dummy' %in% names(inputs)) {
    if (inputs$dummy) x else x
  } else {
    inputs$dummy <<- FALSE
    x
  }}, mutating = TRUE)
```

It's still passing the tests you wrote originally, but I didn't write any additional ones.  Haven't finished going through mungebits2 yet.  Will update you when I do.

### CODE REVIEW

**Comment by davidboren** [comment]
Seeing something weird related to my pre-allocation method when running a test_project().  Gonna find out what it is, then see if I can replicate it in a test.  Hold off on merger.

**Comment by davidboren** [comment]
Okay.  Best I can tell test_project was passing in something that resulted in cols having a length zero.  Added one line of code to check for that case.  All good now as far as I can tell.

**Comment by justinhou** [comment]
@robertzk can you review?

**Comment by robertzk** [comment]
This is great! So still non-linear for some reason, but at least the difference between mutating v.s. immutable transformations is negligible.

File: DESCRIPTION [hunk 1]
```diff
@@ -2,7 +2,7 @@ Package: mungebits
 Type: Package
 Title: mungebits (http://github.com/robertzk/mungebits)
 Description: Mungebits is a collection of data preparation functions
-Version: 0.3.12
+Version: 0.3.12.3
```
Comment (robertzk) [inline DESCRIPTION]: Go ahead and up it to 0.3.13, this is a pretty significant change.

**Comment by FeiYeYe** [comment]
:+1:

**Pull request merged**